pymeasure/instruments/keithley/keithley2306.py

```diff
@@ -0,0 +1,692 @@
+        """A floating point property that controls the integration time (in 
+        seconds) for average pulse measurements. Takes on values between 
+        33.33333e-06 and 0.8333. """,
+        validator=truncated_range,
+        values=[33.33333e-06, 0.8333],
+    )
+
+    pulse_current_time_digitize = Instrument.control(
+        ":SENS<ch>:PCUR:TIME:DIG?", ":SENS<ch>:PCUR:TIME:DIG %g",
+        """A floating point property that controls the integration time (in 
+        seconds) for digitizing or burst pulse measurements. Takes on values 
+        between 33.33333e-06 and 0.8333. """,
+        validator=truncated_range,
+        values=[33.33333e-06, 0.8333],
+    )
+
+    pulse_current_fast_enabled = Instrument.control(
+        ":SENS<ch>:PCUR:FAST?", ":SENS<ch>:PCUR:FAST %d",
+        """A boolean property that controls whether pulse current fast readings 
+        are enabled. """,
+        validator=strict_discrete_set,
+        values={True: 1, False: 0},
+        map_values=True
+    )
+
+    pulse_current_search_enabled = Instrument.control(
+        ":SENS<ch>:PCUR:SEAR?", ":SENS<ch>:PCUR:SEAR %d",
+        """A boolean property that controls whether pulse current search 
+        is enabled. """,
+        validator=strict_discrete_set,
+        values={True: 1, False: 0},
+        map_values=True
+    )
+
+    pulse_current_detect_enabled = Instrument.control(
+        ":SENS<ch>:PCUR:DET?", ":SENS<ch>:PCUR:DET %d",
+        """A boolean property that controls whether pulse current detection 
+        mode is enabled. """,
+        validator=strict_discrete_set,
+        values={True: 1, False: 0},
+        map_values=True
+    )
+
+    pulse_current_timeout = Instrument.control(
+        ":SENS<ch>:PCUR:TOUT?", ":SENS<ch>:PCUR:TOUT %g",
+        """A floating point property that controls the pulse current timeout
+        in seconds, which takes on values between 0.005 and 32. """,
+        validator=truncated_range,
+        values=[0.005, 32],
+    )
+
+    long_integration_trigger_edge = Instrument.control(
+        ":SENS<ch>:LINT:TEDG?", ":SENS<ch>:LINT:TEDG %s",
+        """A string property that controls the long integration trigger edge,
+         which can take the values 'rising', 'falling', or 'neither'. """,
+        validator=strict_discrete_set,
+        values={'rising': 'RISING', 'falling': 'FALLING', 'neither': 'NEITHER'},
+        map_values=True,
+    )
+
+    long_integration_time = Instrument.control(
+        ":SENS<ch>:LINT:TIME?", ":SENS<ch>:LINT:TIME %g",
+        """A floating point property that controls the long integration time
+        in seconds, which takes on values in the range of 0.850 for 60 Hz and
+         0.840 for 50 Hz up to 60. """,
+        validator=truncated_range,
+        values=[0.840, 60],
+    )
+
+    def long_integration_time_auto(self):
+        """Arranges for the instrument to control integration times. """
+        self.write(":SENS<ch>:LINT:TIME:AUTO")
```

Comment:
I think you could avoid this method by using the `joined_validator`  in property `long_integration_time`

Reply:
If I understand the suggestion, this would allow clients write either

```python
instr.ch(1).long_integration_time = 30
```
or
```python
instr.ch(1).long_integration_time = 'AUTO'
```

where the former would result in this SCPI being issued

```
:SENS1:LINT:TIME 30
```

and the latter involving this SCPI

```
:SENS1:LINT:TIME:AUTO
```

An aspect to consider, though, is that there is a similar `pulse_current_time_auto` for automatically setting pulse current measurement parameters. Where long integration involves a single property, pulse current mode involves three distinct properties (`pulse_current_time_high`, `pulse_current_time_low`, and `pulse_current_time_average`).

So a question would be whether allowing a user to set any one of the three pulse current properties to `'AUTO'`, detecting this case, and  make it issue `:SENS<ch>:PCUR:TIME:AUTO` (thus effectively having it automatically set all three), or alternatively leaving things as-is, with methods for each "automatic" setting, adhering a bit more closely to the underlying SCPI exposed by the instrument.

Reply:
Regarding your first comment on `long_integration_time`, yes, indeed this what I meant, but looking at your explanation, I am no longer sure that my suggestion would be applicable, since:

1. The type of the two kind of parameters is different (`str `and `float`), so I am not sure which specifier would go in `":SENS<ch>:LINT:TIME %g"`
2. The syntax is a little bit different, if the parameter is `float`, this is indeed a parameter for SCPI command `:SENS1:LINT:TIME` while in the second case is a different command `:SENS1:LINT:TIME:AUTO`

So, given the above and based on my knowledge on the code and in python, I would leave thing as they are.

Regarding the second question (`pulse_current_time_auto`, sorry I did not spot this), I definitively agree to leave the things as they are.